wizard/archive_purchase_order.py

```diff
@@ -0,0 +1,24 @@
+from odoo import api, models, fields
+from odoo.exceptions import UserError
+
+
+class ArchivePurchaseOrder(models.TransientModel):
+    _name = 'archive.purchase.order'
+    _description = 'Archive Multiple PO'
+
+    archive_purchase_order_ids = fields.Many2many(
+        'purchase.order', string="Purchase Orders")
+    button_archive_invisible = fields.Boolean(
+        compute="_compute_button_invisible")
+
+    @api.depends('archive_purchase_order_ids')
+    def _compute_button_invisible(self):
+        self.button_archive_invisible = False if len(
+            self.archive_purchase_order_ids) > 0 else True
+
+    def action_archive(self):
+        if len(self.archive_purchase_order_ids.filtered(lambda record: record.state not in ['done', 'cancel'])) > 0:
+            raise UserError(
+                'You cannot archive Purchase Orders that are not done/cancelled!')
+        for record in self.archive_purchase_order_ids:
+            record.write({'active': False})
```

Comment:
Replace this `for` loop by `self.archive_purchase_order_ids.write({'active': False})`

Reply:
Also, you didn't make this method satisfy the condition `Only Purchase Manager can archive Purchase Order(s)` because later it can be called from elsewhere